### PR TITLE
[Inductor] [Quant] Enable QConv2d Binary int8-mixed-bf16 Lowering

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #112486
* __->__ #112551
* #112550

**Summary**
- PR 6 for enabling Int8-Mixed-BF16 PT2E PTQ Quantization with Inductor https://github.com/pytorch/pytorch/issues/111640.
- Enable the QConv2d Binary int8-mixed-bf16 post grad lowering inside inductor.

**TestPlan**
```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qconv2d
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Comment by leslie-fang-intel** [comment]
Hi @jgong5, please also help to take a look of this PR.

**Review by jgong5 — APPROVED** [review]

**Review by eellison — APPROVED** [review]

**Review by jerryzh168 — APPROVED** [review]

File: torch/_inductor/fx_passes/quantization.py [hunk 1]
```diff
@@ -532,76 +543,100 @@ def __init__(
...
+                patterns,
+                0 if int8_mixed_bf16_with_inplace_add else 1,  # pass_number
+                torch.ops.onednn.qconv2d_pointwise.binary,  # computation_op
+                # Note that for int8-mixed-bf16 and non-inplace add, because we have
+                # q-dq inserted at extra input of add, so the non-inplace add has bf16 and fp32 inputs,
+                # the output dtype will be float32.
+                # For inplace add, there is a extra to_bf16 node at add output, so the fusion pattern has bfloat16 output.
+                torch.bfloat16 if int8_mixed_bf16_with_inplace_add else torch.float32,
+                binary_unary_attr,  # binary_unary_attr
+            )
```
Comment (jerryzh168) [inline torch/_inductor/fx_passes/quantization.py:616]: it might be better just to have two separate patterns instead of multiple if/else

Comment (leslie-fang-intel) [reply torch/_inductor/fx_passes/quantization.py:616]: Thanks and changed.

**Pull request closed without merging**